posthog/warehouse/models/external_data_schema.py

```diff
@@ -509,19 +516,32 @@ def filter_mssql_incremental_fields(columns: list[tuple[str, str]]) -> list[tupl
     return results
 
 
-def filter_mongo_incremental_fields(columns: list[tuple[str, str]]) -> list[tuple[str, IncrementalFieldType]]:
+def filter_mongo_incremental_fields(
+    columns: list[tuple[str, str]], connection_string: str, collection_name: str
+) -> list[tuple[str, IncrementalFieldType]]:
     results: list[tuple[str, IncrementalFieldType]] = []
+    indexed_fields = get_indexes(connection_string, collection_name)
+
     for column_name, type in columns:
+        # Only include fields that have indexes
+        if column_name not in indexed_fields:
+            continue
+
         type = type.lower()
-        if type == "integer":
+        if type == "timestamp":
+            results.append((column_name, IncrementalFieldType.Timestamp))
+        elif type == "integer":
             results.append((column_name, IncrementalFieldType.Integer))
-        # # MongoDB ObjectId can be used for incremental sync based on timestamp
-        # elif column_name == "_id" and type == "string":
-        #     results.append((column_name, IncrementalFieldType.String))
+        elif column_name == "_id" and type == "string":
+            results.append((column_name, IncrementalFieldType.String))
 
     return results
 
 
+def get_mongo_schemas_for_source_type(job_inputs: dict[str, Any]) -> dict[str, list[tuple[str, str]]]:
+    return get_mongo_schemas(MongoSourceConfig.from_dict(job_inputs))
```

Comment:
This function is a bit redundent - I'd just replace the caller with doing `get_mongo_schemas(MongoSourceConfig.from_dict(job_inputs))` directly